src/go/k8s/controllers/redpanda/cluster_controller_configuration.go

```diff
@@ -0,0 +1,275 @@
+package redpanda
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/go-logr/logr"
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources"
+	"github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources/configuration"
+	"github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources/featuregates"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// preCheckConfigurationChange verifies and marks the cluster as needing synchronization (using the ClusterConfigured condition).
+// When this method returns true, the cluster CR is marked, indicating that the target cluster will be eventually synchronized (via reconcileConfiguration).
+func (r *ClusterReconciler) markConfigurationChanged(
+	ctx context.Context,
```

Comment:
I had hard time to understand in what condition the annotation is applied to the config map. Could you elaborate in this godoc why and when the config map annotation plays the role into reconciliation?

From current description I would assume only Cluster CR condition is updated, but there is fancy way to update annotation too.